### PR TITLE
feat: add confirm dialog

### LINKED ISSUE #17: Component: Migrate ConfirmButton (and rename ConfirmDialog)
# Component Name: ConfirmDialog

### Summary description of UI component
Move component from here:
https://github.com/reactioncommerce/reaction/tree/master/imports/client/ui/components/ConfirmButton

### PR DESCRIPTION
**Pull request by mikemurray**
Resolves #17 
Impact: **major**  
Type: **feature**

## Component
- ConfirmDialog
- Add MUI theme

## Screenshots
--

## Breaking changes

none

## Testing
1. See the `ConfirmDialog` docs
2. Open the dialog to test
3. Click "cancel" once
4. Click "ok" once

### CODE REVIEW

**Comment by machikoyasuda** [comment]
Because this is our first component and part of our goals for this pod is to be very thorough with review process and document everything, I'm going to document everything I'm reviewing. And I'm tagging @nnnnat for reviews on accessibility + testing, and @rymorgan for a design review.

As a **code reviewer** of this component, I am reviewing:

## Code
- [ ] Naming: Is the component naming confusing?
- [ ] Code: Linting should pass.
- [ ] Commit messages: Make sure the `fix` and `feat`s are correct.
- [ ] Component: Are the ARIA labels, etc. compliant (I dunno how to check this...yet!)

## Tests
- [ ] Tests: Are there enough test cases to cover the major (and minor) uses of this component?
- [ ] Tests: All tests shouuld pass.

## Docs
- [ ] Documentation: Is it clear how to use this component?
- [ ] Documentation: Is the component added to the sidebar? To the correct section of the sidebar?

## Manual testing:
- [ ] Component: Is it mobile-first? Check on a phone.
- [ ] Component: Is the component usable with only a keyboard?

@mikemurray Can you think of any other things we should be reviewing for when reviewing components as a code reviewer?

**Comment by machikoyasuda** [comment]
@rymorgan stuff to review for

- [ ] Does the naming of the component overall look good?
- [ ] Is the component in the right sidebar section?
- [ ] What other docs should we add? What other use cases?

File: package/src/components/ConfirmDialog/ConfirmDialog.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+### Overview
+
```
Comment (machikoyasuda) [inline package/src/components/ConfirmDialog/ConfirmDialog.md]: Would it be useful to have dev docs here like this about where to find the corresponding MUI docs?

Example for this component:

`The ConfirmDialog component uses [Dialog](https://material-ui.com/api/dialog), DialogActions, DialogContent, DialogContentText, DialogTitle and Button from Material-UI.`

**Comment by machikoyasuda** [comment]
This PR also resolves https://github.com/reactioncommerce/catalyst/issues/21

File: package/src/components/ConfirmDialog/ConfirmDialog.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+### Usage
+
+The `ConfirmDialog` component provides a standard confirmation alert base on some user action.
```
Comment (machikoyasuda) [inline package/src/components/ConfirmDialog/ConfirmDialog.md]:
```suggestion
The `ConfirmDialog` component provides a standard confirmation alert base on some user action. Pass a callback function to `onConfirm()` to allow the user to take an action after clicking the confirmation button.
```

**Comment by machikoyasuda** [comment]
## Actions to be taken in `reaction` after this component is merged + deployed

1. Replace all instances of `ConfirmDialog` and `ConfirmButton` in `reaction` with this Catalyst one. Test them.
1. @mikemurray What needs to be done with the `imports/plugins/core/client/theme/muiTheme.js` file after this PR is merged, if anything? Can we move this file to a place that makes more sense, like `imports/client/ui/components/`?

**Comment by machikoyasuda** [comment]
#27 merged into master. You'll have to pull from master and do some git backflips! 🤸‍♂

**Comment by mikemurray** [comment]
@machikoyasuda An additional thing to look out for when testing would be the following

- [ ] Ensure the component is forwarding refs.

This just ensures that when refs are used, the correct component is referenced. This is especially true for our Catalyst/MUI components so they retain similarly functionality.

Basic example: 

```js
const ConfirmDialog = React.forwardRef(function ConfirmDialog(props, ref) {
  return <Dialog ref={ref}></Dialog>
});
```

**Comment by mikemurray** [comment]
@machikoyasuda @rymorgan ready for a re-review

**Pull request merged**